src/NServiceBus.Core.Analyzer/FeatureDefaultsEnableFeatureAnalyzer.cs

```diff
@@ -0,0 +1,122 @@
+namespace NServiceBus.Core.Analyzer
+{
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class FeatureDefaultsEnableFeatureAnalyzer : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(diagnostic);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(startContext =>
+            {
+                var featureType = startContext.Compilation.GetTypeByMetadataName("NServiceBus.Features.Feature");
+                var settingsExtensionsType = startContext.Compilation.GetTypeByMetadataName("NServiceBus.Features.SettingsExtensions");
+
+                if (featureType == null || settingsExtensionsType == null)
+                {
+                    return;
```

Comment:
Open question, I have not myself determined if it's better in situations like this to return null or to throw an exception. If it fails it means something serious is wrong like we've changed the full name of the Feature class (unlikely) so would we want it to "break more"? Although the other side of that argument is that tests should fail if it just returns too and that would insulate a user from getting a more serious editor in an even more crazy assembly load situation.

Reply:
Yeah I'm fine with it, just wanted to raise the thought.